Recon/src/Recon/SequentialTrigger.cxx

```diff
@@ -1,54 +1,64 @@
-
 #include "Recon/SequentialTrigger.h"
 #include "Recon/Event/TriggerResult.h"
 
 namespace recon {
 
+/**
+ *Instantiates the variables used in this processor. We take trigger_list_ and tigger_passNames_ to obtain trigger collections, a doOR_ and doAND_ option to check if one of or all events pass, and doVAL_ to output a validation collection
+ * */
+
 void SequentialTrigger::configure(framework::config::Parameters &ps) {
   trigger_list_ = ps.getParameter<std::vector<std::string>>("trigger_list");
   trigger_passNames_ = ps.getParameter<std::vector<std::string>>("trigger_passNames");
-  pass_masks_ = ps.getParameter<std::vector<int>>("pass_mask");
   doAND_ = ps.getParameter<bool>("doAND");
   doOR_ = ps.getParameter<bool>("doOR");
+  doVAL_ = ps.getParameter<bool>("doVAL");
   return;
 }
 
+/**
+ *
+ *This producer takes in a list of triggers and runs an OR or AND skimmer depending on the config file. It will change the keep
+ tag and produce nothing. This is unless the validation command is set to true, in which case it produces a boolean collection in an output file
+ with the keeping flag.
+ *
+ * */
+
 void SequentialTrigger::produce(framework::Event& event) {
   bool hasPassed = false;
   bool keepTrack = false;
-  int  maskValue = 0; 
+  //Returns an error if some bad combination of doOR and doAND is enabled.
+  if (doAND_ and doOR_){
+    ldmx_log(fatal) << "Can't do both doAND and doOR. Exiting.";
+    return;
+  }
+  if (not(doAND_) and not(doOR_)){
+    ldmx_log(fatal) << "Must do one of doAND and doOR. Exiting.";
+    return;
+  }
```

Comment:
if doOR == doAND...

Reply:
but of course you get less granular error messages that way

Reply:
also move these lines to the `configure` method - you could even simply throw an exception with `EXCEPTION_RAISE(<short-code>,<message>);` since it doesn't make sense to run this processor without choosing one of these options.